rocAL/rocAL/source/meta_data/coco_meta_data_reader.cpp

```diff
@@ -142,9 +157,178 @@ void COCOMetaDataReader::print_map_contents()
+    std::unique_ptr<uint[]> cnts = std::make_unique<uint[]>(h * w + 1);
+    std::unique_ptr<int[]> vals = std::make_unique<int[]>(h * w + 1);
+    for (auto label = ++labels.begin(); label != labels.end(); label++) {
+        RLE B = R[*label];
+        if (B.cnts == 0)
+            continue;
+
+        uint cnt_a = A.cnts[0];
+        uint cnt_b = B.cnts[0];
+        int next_val_a = A.vals[0];
+        int val_a = next_val_a;
+        int val_b = *label;
+        bool next_vb = false;
+        bool vb = next_vb;
+        uint nb_seq_a, nb_seq_b;
+        nb_seq_a = nb_seq_b = 1;
+        int m = 0;
+
+        int cnt_tot = 1;  // check if we advanced at all
+        while (cnt_tot > 0) {
+            uint c = std::min(cnt_a, cnt_b);
+            cnt_tot = 0;
+            // advance A
+            cnt_a -= c;
+            if (!cnt_a && nb_seq_a < A.m) {
+                cnt_a = A.cnts[nb_seq_a];  // next sequence for A
+                next_val_a = A.vals[nb_seq_a];
+                nb_seq_a++;
+            }
+            cnt_tot += cnt_a;
+            // advance B
+            cnt_b -= c;
+            if (!cnt_b && nb_seq_b < B.m) {
+                cnt_b = B.cnts[nb_seq_b++];  // next sequence for B
+                next_vb = !next_vb;
+            }
+            cnt_tot += cnt_b;
+
+            if (val_a && vb)  // there's already a class at this pixel
+                            // in this case, the last annotation wins (it's undefined by the spec)
+                vals[m] = (!cnt_a) ? val_a : val_b;
+            else if (val_a)
+                vals[m] = val_a;
+            else if (vb)
+                vals[m] = val_b;
+            else
+                vals[m] = 0;
+            cnts[m] = c;
+            m++;
+
+            // since we switched sequence for A or B, apply the new value from now on
+            val_a = next_val_a;
+            vb = next_vb;
+
+            if (cnt_a == 0) break;
+        }
+        // copy back the buffers to the destination encoding
+        A.m = m;
+        for (int i = 0; i < m; i++) A.cnts[i] = cnts[i];
+        for (int i = 0; i < m; i++) A.vals[i] = vals[i];
+    }
+
+    // Decode final pixelwise masks encoded via RLE and polygons
+    memset(pixelwise_labels.data(), 0, h * w * sizeof(int));
+    int x = 0, y = 0;
+    for (uint i = 0; i < A.m; i++) {
+        for (uint j = 0; j < A.cnts[i]; j++) {
+            pixelwise_labels[x + y * w] = A.vals[i];
+            if (++y >= h) {
+                y = 0;
+                x++;
+            }
+        }
+    }
+
+    // Destroy RLEs
+    rlesFree(&R, *labels.rbegin() + 1);
+    for (auto rles : frPoly)
+        for (auto &rle : rles.second)
+            rleFree(&rle);
+}
+
 void COCOMetaDataReader::read_all(const std::string &path)
 {
     _coco_metadata_read_time.start(); // Debug timing
+    std::string rle_str;
+    std::vector<uint32_t> rle_uints;
+    uint32_t max_width=0, max_height=0;
+    RLE *R = (RLE*) malloc(sizeof(RLE));
```

Comment:
I feel it should be malloc because this it is passes as input for third party param
Unique pointer can be done with custom deleter,
https://stackoverflow.com/questions/3477525/is-it-possible-to-use-a-c-smart-pointers-together-with-cs-malloc
What you think on it?